include/core/shape.h

```diff
@@ -899,11 +912,12 @@ class CompositeShape : public Shape<dim>
     , constituents(constituents)
     , operations(operations)
   {
-    // Calculation of the effective radius
-    for (auto const &[component_id, component] : constituents)
+    // Calculation of the effective radius and setting of constituents' status
+    for (auto const &[constituent_id, constituent] : constituents)
```

Comment:
You could use a boost::value iterator adaptor to not have to declare the pair like this unused.